hubblestack/log.py

```diff
@@ -154,10 +158,14 @@ def setup_splunk_logger():
     SPLUNK_HANDLER = handler
 
 
-def emit_to_splunk(message, level, name):
+def emit_to_splunk(message, level, name, remove_sensitive_logs=False):
     '''
     Emit a single message to splunk
     '''
+
+    if isinstance(message, (list, dict)):
+        message = filter_logs(message, remove_dots=False)
```

Comment:
Note that I have changed the default value of  "remove_dots" argument to False.

Firstly, I don't want to remove dots from “hubblestack.grains_report” logs. I fear that it could break people's splunk search queries .

Plus I had some other doubts:

Q. Why didn't I add another parameter in the signature of "emit_to_splunk" for "remove_dots"?
A. I don't want to complicate the signature of emit_to_splunk further.  "remove_dots" won't make any sense if "remove_sensitive_logs" is False.

Q. What if someone want's to remove_dots also ?
A. IN that case he should call "filter_logs" first ( with  remove_dots=True ). Then he can call "emit_to_splunk" (  with remove_sensitive_logs=False as the logs have already been filtered )

Q.  Before this commit ( when remove_dots was set to True ) , could we cover the scenario where someone wants to filter the logs but don't want to remove dots ?
A. Nope. That's why I changed it to remove_dots=False. Now one can cover all the scenarios by mix and match of "emit_to_splunk" and "filter_logs"


I know, not very strong reasoning. But I had to go with atleast one of the choices. 
Let me know if someone thinks otherwise.

Reply:
Our reasoning is for conf_publisher, not grains report. There are some settings with dots in the config -- the ones that come to mind are for s3 credentials (s3.key, s3.keyid, etc). The problem with this is that searching nested data structures in splunk is dot-based. So if you want to search for those keys, it becomes very difficult and you have to deal with escaping and it makes life hard.

I am going to review this further and maybe make it a conf_publisher argument instead.

Reply:
Oh, we already do that. We're good! This should be fine.